libs/langgraph/src/agent/middleware/humanInTheLoop.ts

```diff
@@ -0,0 +1,120 @@
+import { AIMessage, isAIMessage, ToolMessage } from "@langchain/core/messages";
+import { AgentMiddleware, AgentState } from "../agent";
+import { interrupt } from "../../interrupt";
+import {
+  HumanInterruptConfig,
+  ActionRequest,
+  HumanInterrupt,
+  HumanResponse,
+} from "../../prebuilt/interrupt";
+
+export type ToolInterruptConfig = Record<string, HumanInterruptConfig>;
+
+export class HumanInTheLoopMiddleware implements AgentMiddleware {
```

Comment:
Do you have an example here? Let's say we have a GitHub triage agent that we trigger and requests approval of the triage result before posting a comment on the issue. How would that be implemented from end-to-end?